packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2366,6 +2440,7 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     } else {
       currentSelectionStartIndex = newIndex;
     }
+    _flushInactiveSelections();
```

Comment:
just something to keep in mind, we know which selectables currently have selection after the adjustment, we need to go through the entire selectable list. This is fine as long as the performance is fine.